src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -196,19 +196,19 @@ public CreatePaymentResponse initiatePayment(
 
         Claim createdClaim = caseRepository.initiatePayment(user, claim);
 
+        Optional<Payment> payment = createdClaim.getClaimData().getPayment();
         return CreatePaymentResponse.builder()
-            .nextUrl(createdClaim.getClaimData().getPayment().getNextUrl())
+            .nextUrl(payment.orElseThrow(IllegalStateException::new).getNextUrl())
```

Comment:
May as well throw the exception as soon as we have the optional

Reply:
`payment` is being initialised in call back now not coming from front-end.